org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/Bean.java

```diff
@@ -0,0 +1,157 @@
+ * able to load the class referenced and set the fields using reflection.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class Bean {
+	
+	private String             beanClass;
+    private Map<String,Object> data;
+	
+	public Bean() {
+		this(null);
+	}
+	
+	public Bean(String beanClass) {
+		this.beanClass = beanClass;
+		this.data      = new HashMap<>();
+	}
+	
+	/**
+	 * Shortcut method used for when you already have an instance of the bean.
+	 * @param theBean
+	 * @throws NoSuchMethodException
+	 * @throws SecurityException
+	 * @throws IllegalAccessException
+	 * @throws IllegalArgumentException
+	 * @throws InvocationTargetException
+	 */
+	public Bean(Object theBean) throws NoSuchMethodException, 
+										SecurityException, 
+										IllegalAccessException, 
+										IllegalArgumentException, 
+										InvocationTargetException {
+		
+		this.beanClass = theBean.getClass().getName();
+		this.data      = new HashMap<>();
+		init(theBean);
+	}
+
+	/**
+	 * Set a field value of the mapped beans.
+	 * @param fieldName
+	 * @param value
+	 */
+	public <T> void set(String fieldName, T value) {
+		data.put(fieldName, value);
+	}
+	
+	@SuppressWarnings("unchecked")
+	public <T> T get(String fieldName) {
+		return (T)data.get(fieldName);
+	}
+	
+	/**
+	 * 
+	 * @return a Set of the field names
+	 */
+	public Collection<String> names() {
+		return new HashSet<String>(data.keySet());
+	}
+
+	public String getBeanClass() {
+		return beanClass;
+	}
+
+	public void setBeanClass(String beanClass) {
+		this.beanClass = beanClass;
+	}
+
+	public Map<String, Object> getData() {
+		return data;
+	}
+
+	public void setData(Map<String, Object> data) {
+		this.data = data;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((beanClass == null) ? 0 : beanClass.hashCode());
+		result = prime * result + ((data == null) ? 0 : data.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Bean other = (Bean) obj;
+		if (beanClass == null) {
+			if (other.beanClass != null)
+				return false;
+		} else if (!beanClass.equals(other.beanClass))
+			return false;
+		if (data == null) {
+			if (other.data != null)
+				return false;
+		} else if (!data.equals(other.data))
+			return false;
+		return true;
+	}
+	
+    /**
+     * Method reflects the fields into the map of field values.
+     * @param theBean
+     * @throws NoSuchMethodException
+     * @throws SecurityException
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
+     * @throws InvocationTargetException
+     */
+	private void init(Object theBean) throws NoSuchMethodException, SecurityException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.SecurityException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)